Add support for categories

georgehrke opened the pull request:
fixes #107 

ToDo:
- [x] read only view is missing

![6CF201B5-1891-4898-8984-55D3786F2B33](https://user-images.githubusercontent.com/1250540/67214532-33ff2200-f420-11e9-9d85-b57418ec37cf.png)

![B98F2439-E10D-459F-B4AC-D0A387885A9A](https://user-images.githubusercontent.com/1250540/67214571-3d888a00-f420-11e9-9ae6-16a6d3a76945.png)

georgehrke commented:
Requires: https://github.com/georgehrke/calendar-js/pull/38

georgehrke commented:
I also added an indicator in the calendar-grid view, but I'm not 100% sure about that.

![3BDCD0F7-7D77-4EB8-A9D1-68E29FDAE97B](https://user-images.githubusercontent.com/1250540/67218398-9a873e80-f426-11e9-863c-82b139e9271d.png)

georgehrke commented:
@tcitworld What do you think about the indicator in the calendar-grid?

georgehrke commented on src/components/Editor/Properties/PropertySelectMultiple.vue:
```diff
@@ -0,0 +1,117 @@
+			return this.propModel.options
+		},
+		foo() {
+			return uidToColor('abc')
```
Remove me

tcitworld commented:
Can we ask to move @skjnldsv's uidToColor.js into a common `@nextcloud` package?

georgehrke commented:
> Can we ask to move @skjnldsv's uidToColor.js into a common @nextcloud package?

Yes. Please. We were wondering earlier whether to put it into a dedicated package or just nextcloud/vue

georgehrke commented:
(Damn, npm naming, sorry for mentioning everyone)

tcitworld commented:
> What do you think about the indicator in the calendar-grid?

I've never seen any indicator apart from the thunderbird screenshot in the issue, so I'm not sure how much it would be useful.

Maybe we should get it consistent with https://github.com/nextcloud/server/issues/13330 and https://github.com/nextcloud/server/issues/6778?

skjnldsv commented:
> Yes. Please. We were wondering earlier whether to put it into a dedicated package or just nextcloud/vue

Currently it's already in @nextcloud/vue :)

jancborchardt commented:
With so many full-on colors, it probably makes sense to maybe look into using a similar way of coloring as macOS Calendar, with the events by default not being full-saturation (but only the active one):
![](https://help.apple.com/assets/5C59FB23680CE2D15FFB5957/5C59FB26680CE2D15FFB595E/en_GB/1059edf44a96a1c7c9e587b028e83555.png)

More: https://support.apple.com/en-euro/guide/calendar/welcome/mac

georgehrke commented:
@jancborchardt How is that related to categories? :)

jancborchardt commented:
Sorry, that was in reference to your comment above at https://github.com/nextcloud/calendar/pull/1540#issuecomment-544563314

So basically I meant to say: Let’s not put any indicators in the view for now. We have more stuff to clean up before we put stuff in again. :)

georgehrke commented:
I removed the indicator in the calendar-grid for now and added a read-only version:
![BC208BD7-FBE3-48BF-990F-A45859A261B8](https://user-images.githubusercontent.com/1250540/67618758-d38f2c80-f7f3-11e9-88ff-5c568f0bc9e4.png)
